File: chazm-model/src/main/java/io/github/runtimemodels/chazm/event/ContainsEventFactory.java
package io.github.runtimemodels.chazm.event;

import io.github.runtimemodels.chazm.relation.Contains;

/**
 * The {@linkplain ContainsEventFactory} interface defines the API for constructing {@linkplain ContainsEvent}s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
public interface ContainsEventFactory {

    /**
     * Constructs an {@linkplain ContainsEvent}.
     *
     * @param category the {@linkplain EventCategory}.
     * @param contains the {@linkplain Contains}.
     * @return a {@linkplain ContainsEvent}.
     */
    ContainsEvent build(EventCategory category, Contains contains);

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/relation/ContainsEventFactory.kt
package runtimemodels.chazm.model.relation

import runtimemodels.chazm.api.relation.Contains
import runtimemodels.chazm.model.event.EventType

/**
 * The [ContainsEventFactory] interface defines the API for constructing [ContainsEvent]s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
interface ContainsEventFactory {
    /**
     * Constructs an [ContainsEvent].
     *
     * @param category the [EventType].
     * @param contains the [Contains].
     * @return a [ContainsEvent].
     */
    fun build(category: EventType, contains: Contains): ContainsEvent
}
